File: nodejs/src/entities/Persona.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, OneToMany } from 'typeorm';
import { PersonaAtributo } from './PersonaAtributo';

@Entity('personas')
export class Persona {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'varchar', length: 200 })
  nombreCompleto: string;

  @Column({ type: 'varchar', length: 20, unique: true })
  identificacion: string;

  @Column({ type: 'int' })
  edad: number;

  @Column({ type: 'varchar', length: 10 })
  genero: string;

  @Column({ type: 'varchar', length: 20, default: 'Activo' })
  estado: string;

  @Column({ type: 'nvarchar', length: 'MAX', default: '{}' })
  atributosAdicionales: string;

  @CreateDateColumn()
  fechaCreacion: Date;

  @UpdateDateColumn({ nullable: true })
  fechaModificacion?: Date;

  @OneToMany(() => PersonaAtributo, personaAtributo => personaAtributo.persona, { cascade: true })
  personaAtributos: PersonaAtributo[];
}

File: dotnet/PersonasABM.Domain/Entities/Persona.cs
using System.ComponentModel.DataAnnotations;

namespace PersonasABM.Domain.Entities;

public class Persona
{
    public int Id { get; set; }
    
    [Required]
    [StringLength(200)]
    public string NombreCompleto { get; set; } = string.Empty;
    
    [Required]
    [StringLength(20)]
    public string Identificacion { get; set; } = string.Empty;
    
    [Range(1, 150)]
    public int Edad { get; set; }
    
    [Required]
    [StringLength(10)]
    public string Genero { get; set; } = string.Empty;
    
    [Required]
    [StringLength(20)]
    public string Estado { get; set; } = "Activo";
    
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    
    public DateTime? FechaModificacion { get; set; }
    
    // Atributos adicionales como JSON
    public string AtributosAdicionales { get; set; } = "{}";
    
    // Navegación
    public virtual ICollection<PersonaAtributo> PersonaAtributos { get; set; } = new List<PersonaAtributo>();
}
